### PR TITLE
Making experimental package public and updating its dependency versions

### PR DESCRIPTION
**Pull request by louwie17**
Making the experimental package public as our components package and navigation package depend on it.
I also changed the explat package to private, so I can use `lerna` for publishing. 

### Detailed test instructions:

Check changelogs and version numbers to see if all looks good.

No changelog necessary.

### CODE REVIEW

**Review by samueljseay — DISMISSED** [review]
As we discussed in other channels I don't think changing the file link is a good idea here. Lets talk to @psealock about next steps but I think we want to publish the 1.0 of @woocommerce/experimental for now.

File: packages/explat/package.json [hunk 1]
```diff
@@ -32,5 +32,6 @@
 	},
 	"devDependencies": {
 		"@types/cookie": "^0.4.0"
-	}
+	},
+	"private": true
```
Comment (psealock) [inline packages/explat/package.json:36]: Lets go ahead and publish this version of `explat`

**Review by psealock — COMMENTED** [review]
Thanks @louwie17 for handling this. Everything is looking good, except lets publish explat as well. There are a couple more additional changes:

* Add `@woocommerce/experimental` and `@woocommerce/explat` to DEWP https://github.com/woocommerce/woocommerce-admin/blob/main/packages/dependency-extraction-webpack-plugin/assets/packages.js
* Add `experimental` to webpack so a folder is created in `dist/`

**Comment by psealock** [comment]
Here is a summary of changes I made in this PR. 

First, a note about using filepaths instead of explicit version numbers to reference packages depending on other package: Filepaths is more correct, but in the end this change didn't make a difference for WC Admin because that code is being gathered off the `window.wc` object and not whats in `node_modules`. I went ahead and changed it to avoid future confusion.

* Made `explat` public and added `experimental` to Webpack's build https://github.com/woocommerce/woocommerce-admin/pull/6810/commits/275a3f8aca1bbd455cd63235f69850499588f824
* Bumped versions of Dependency Extraction Webpack Plugin to use new packages in https://github.com/woocommerce/woocommerce-admin/pull/6810/commits/c0e235f55f1c617a7684a4ab97dd3d00dd3dc862
* Made packages depending on other packages use file paths instead of explicit version numbers in https://github.com/woocommerce/woocommerce-admin/pull/6810/commits/89031d0b33f26e55b708c2679bbb7e470f2bbd76
* Made sure the `wc-experimental` script is registered in Loader.php https://github.com/woocommerce/woocommerce-admin/pull/6810/commits/1d058d5649e6682cacb569817ba5bd0ad1d15fdd
* Removed an unneeded check to exclude experimental package in https://github.com/woocommerce/woocommerce-admin/pull/6810/commits/5d98b7771553b62b2f1f2bd5bbe5c0da67b7f46b

## How I tested this

I ran `npm start` and made sure `window.wc.explat` and `window.wc.experimental` were defined. 

This tells me:
* dependency-extraction-webpack-plugin is correctly identifying these dependencies
* Loader.php is correctly registering the scripts, 
* The dependencies themselves are being loaded
* Webpack knows not to bundle these assets

Next, I needed to know if an external plugin could use these assets without error. To do this, I need to `npm link` dependency-extraction-webpack-plugin because changes in that package have not yet reached npm, so local copy must be used.

* `npm run build`
* `cd packages/dependency-extraction-webpack-plugin`
* `npm link`
* `cd ../../../other-extension`
* `npm link @woocommerce/dependency-extraction-webpack-plugin`
* `npm start`

In my extension, this code executes without error

```js
/**
 * External dependencies
 */
import { Text } from '@woocommerce/experimental';
import { SummaryList } from '@woocommerce/components';
import { Experiment } from '@woocommerce/explat';

console.log( Text );
console.log( SummaryList );
console.log( Experiment );
```

**Review by samueljseay — APPROVED** [review]
🚀

**Pull request merged**